src/lib/caching/structures/presences/PresenceGameBuilder.ts

```diff
@@ -0,0 +1,271 @@
+	 */
+	public setCreatedAt(createdAt: number): this {
+		this.created_at = createdAt;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param timestamps Unix timestamps for start and/or end of the game.
+	 */
+	public setTimestamps(timestamps: APIActivityDataTimestamp): this {
+		this.timestamps = timestamps;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param applicationID Application id for the game.
+	 */
+	public setApplicationID(applicationID: string): this {
+		this.application_id = applicationID;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param details What the player is currently doing.
+	 */
+	public setDetails(details: string | null): this {
+		this.details = details;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param state The user's current party status.
+	 */
+	public setState(state: string | null): this {
+		this.state = state;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param emoji The emoji used for a custom status.
+	 * @see https://discord.com/developers/docs/topics/gateway#activity-object-activity-emoji
+	 */
+	public setEmoji(emoji: APIActivityDataEmoji | null): this {
+		this.emoji = emoji;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param party Information for the current party of the player.
+	 * @see https://discord.com/developers/docs/topics/gateway#activity-object-activity-party
+	 */
+	public setParty(party: APIActivityDataParty): this {
+		this.party = party;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param assets Images for the presence and their hover texts.
+	 * @see https://discord.com/developers/docs/topics/gateway#activity-object-activity-assets
+	 */
+	public setAssets(assets: APIActivityDataAssets): this {
+		this.assets = assets;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param secrets Secrets for Rich Presence joining and spectating.
+	 * @see https://discord.com/developers/docs/topics/gateway#activity-object-activity-secrets
+	 */
+	public setSecrets(secrets: APIActivityDataSecrets): this {
+		this.secrets = secrets;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param instance Whether or not the activity is an instanced game session.
+	 */
+	public setInstance(instance: boolean): this {
+		this.instance = instance;
+		return this;
+	}
+
+	/**
+	 * Modifies this presence game and returns it.
+	 * @since 0.0.1
+	 * @param flags Activity flags `OR`d together, describes what the payload includes.
+	 */
+	public setFlags(flags: ActivityFlags): this {
```

Comment:
So..we have to do the bit calculations ourselves? I thought we had a class for this

Reply:
No, we don't, `ActivityFlags` is only referenced in this file.
Yes, I could make it a bitfield class.

Reply:
Actually, we do. Pushing a fix.

Reply:
Fixed in b7d7e5a! 🎉